opensearchpy/_async/transport.py

```diff
@@ -116,8 +118,13 @@ def __init__(
         self._async_init_called = False
         self._sniff_on_start_event = None  # type: asyncio.Event
 
+
         super(AsyncTransport, self).__init__(
-            hosts=[],
+            hosts=self.__ensure_compatability_with_legacy_test_cases(
```

Comment:
This strikes me as something in non-test code to support test code, which is probably not right. Is there a legit (non-test) use-case where `hosts` should not be passed down? If not, tests should be modified to ensure that's always the case.

Reply:
@dblock Hey! Texting to you after deeper investigations.

**In summary**: it was my misunderstanding of connections management life-cycle within `AsyncTransport` & everything is fine from `opensearch-py` side.

## More details
- Right after creation of connection instance inside `AsyncTransport`, the sniffing operation is performed as a verification that connection is valid
- meanwhile the actual instantiation of connection occurs not in `AsyncTransport.__init__()`, but inside of `AsyncTransport._async_init()` which is called mostly outside of the cluster
- I suppose the intution behind such implementation approach was setup capabilities for lazy initialization

## Actions from my side
- I've added test cases demonstrating population of connection pool by connectors
- under **NOTE** section of the test cases i left some small explanation

Thanks for your time and assistance!